Microsoft.DotNet.Interactive/Utility/DirectoryUtility.cs

```diff
@@ -85,5 +86,21 @@ private static DirectoryInfo EnsureExists(this DirectoryInfo directory)
 
             return directory;
         }
+
+        public static async Task<bool> WaitForFiles(this DirectoryInfo directory, TimeSpan timeout, Func<FileInfo[], bool> trigger)
```

Comment:
It would be better to accept a `CancellationToken` and throw if it elapses.

Reply:
Given the way it's being used you might also change the return type to a `FileInfo` and return the specific file.

Reply:
Honestly, all of this is a hack to see if I can get the test to consistently pass in CI.  Once I can get it to pass I'll make it look nice.  Naturally it works perfectly on my local machine.